fecgov/fec-cms
# Fix prototype pollution vulnerability

## Summary (required)
Update datatables.net-responsive and datatables.net packages 

- Resolves #4325 


_Include a summary of proposed changes._

1.   In package.json update:
       -  datatables.net to v1.10.23,
       -  datatables.net-responsive": 2.2.7
2.    Update `datatables.net-responsive` dependencies in package-lock.json 

## How to test:

- Checkout `develop` branch.  Pull the latest changes and  run `snyk test --file=package.json` . Notice the Prototype pollution vulnerability shows up in the terminal as both HIGH and MEDIUM priority.
<img width="1228" alt="Screen Shot 2021-01-15 at 1 27 22 PM" src="https://user-images.githubusercontent.com/11650355/104764527-83242780-5735-11eb-8e5f-3c00524617b6.png">

- Checkout `feature/4325-fix-prototype-pollution`
- Delete node modules folder:  run `rm -rf node-modules/`
- Install  `datatables.net` and `datatables.net-responsive` packages: run `npm i datatables.net-responsive@latest datatables.net@latest ` and `npm run build`
- run `snyk test --file=package.json` In the terminal,  prototype pollution vulnerability no longer appears (unlike the develop branch, screenshot attached above)

- I deployed a test branch to `dev` space and tested raising, spending, candidate datatable section to make sure there are no errors i the console. Here is the circleci build -https://app.circleci.com/pipelines/github/fecgov/fec-cms/877/workflows/2319cb7d-4181-4886-98e6-c8f64d1f20ae/jobs/4140
